packages/core/src/services/ai/index.ts

```diff
@@ -27,27 +26,16 @@ import { applyAllRules } from './providers/rules'
 import { VercelConfig } from '@latitude-data/constants'
 import { Result } from './../../lib/Result'
 import { TypedResult } from './../../lib/Result'
+import { getLanguageModel } from './getLanguageModel'
 
 const DEFAULT_AI_SDK_PROVIDER = {
   streamText: originalStreamText,
-  streamObject: originalStreamObject,
 }
 type AISDKProvider = typeof DEFAULT_AI_SDK_PROVIDER
-type AIReturnObject = Pick<
-  StreamObjectResult<unknown, unknown, never>,
-  'fullStream' | 'object' | 'usage' | 'providerMetadata'
-> & {
-  type: 'object'
-  providerName: Providers
-}
 
-// A stream of partial outputs. It uses the `experimental_output` specification.
-// This could fix the issue with having a schema and tool calls in the same prompt.
-// But requires more investigation. More info:
-// https://vercel.com/blog/ai-sdk-4-1#structured-output-improvements
```

Comment:
Simplified all of this. Before we were returning eithr `AIReturnObject` or `AIReturnText` and using `streamObject` or `streamText`

I just saw that we have in `streamText` `experimental_output` so we can pass the JSON schema expected output to stream text. 

I think this is better and fix a combination of tools + schema that we were not handling